src/app/plugins/plugin_detect_balls.cpp

```diff
@@ -248,7 +248,9 @@ ProcessResult PluginDetectBalls::process ( FrameData * data, RenderOptions * opt
             for (int r = 0; r < robots_n; r++) {
               const SSL_DetectionRobot & robot = robots->Get(r);
               if (robot.confidence() > 0.0) {
-                if ((sq((double)(robot.x())-(double)(field_pos.x)) + sq((double)(robot.y())-(double)(field_pos.y))) < near_robot_dist_sq) {
+                vector3d field_on_bot_pos_3d;
+                camera_parameters.image2field ( field_on_bot_pos_3d, pixel_pos, robot.height());
```

Comment:
Why is this necessary? It is already done in lines 217-220.

Reply:
the difference is, that here I transform the blob to robot height.
Because if e.g. a magenta blob is falsely detected as orange und thus inside the bot pattern, it is at robot height. The filter did not behave correctly before.

Reply:
Ah, I see. Yes, makes sense.